src/components/FabAccountButtonBar.js

```diff
@@ -0,0 +1,231 @@
+      setNext();
+      navigation.navigate(name, {
+        ...options,
+        params: {
+          ...(options ? options.params : {}),
+          accountId,
+          parentId,
+        },
+      });
+    },
+    [account, parentAccount, navigation],
+  );
+
+  const onPress = useCallback(
+    (data: ActionButtonEventProps) => {
+      const { navigationParams, confirmModalProps, enableActions } = data;
+      if (!confirmModalProps) {
+        setInfoModalProps();
+        if (navigationParams) onNavigate(...navigationParams);
+        if (enableActions) setDisplayedActions(enableActions);
+      } else {
+        setInfoModalProps(data);
+        setIsModalInfoOpened(true);
+      }
+    },
+    [onNavigate, setIsModalInfoOpened],
+  );
+
+  const goToNext = useCallback(() => {
+    if (next) {
+      // workaround for bottom modal + text input autoFocus issue
+      setTimeout(() => onNavigate(...next), 0);
+    }
+  }, [onNavigate, next]);
+
+  const onContinue = useCallback(() => {
+    setIsModalInfoOpened(false);
+    onPress({ ...infoModalProps, confirmModalProps: undefined });
+  }, [infoModalProps, onPress]);
+
+  const onClose = useCallback(() => {
+    setIsModalInfoOpened();
+  }, []);
+
+  return (
+    <View style={styles.root}>
+      {buttons.map(
+        (
+          { label, Icon, event, eventProperties, Component, ...rest },
+          index,
+        ) => (
+          <Button
+            title={label}
+            IconLeft={Icon}
+            event={event}
+            eventProperties={eventProperties}
+            containerStyle={[styles.button, styles.dropShadow]}
+            type="primary"
+            onPress={() => onPress(rest)}
+            key={index}
+          />
+        ),
+      )}
+      {isModalInfoOpened && infoModalProps && (
+        <InfoModal
+          {...(infoModalProps.confirmModalProps
+            ? infoModalProps.confirmModalProps
+            : {})}
+          onContinue={onContinue}
+          onClose={onClose}
+          isOpened={!!isModalInfoOpened}
+        />
+      )}
+
+      {actions && actions.default.length > 0 && (
+        <>
+          <TouchableOpacity
+            style={[
+              styles.moreButton,
+              styles.dropShadow,
+              { backgroundColor: colors.live },
+            ]}
+            onPress={() => setDisplayedActions("default")}
+          >
+            <Ellipsis color="white" size={16} />
+          </TouchableOpacity>
+          <BottomModal
+            isOpened={!!displayedActions}
+            onClose={() => setDisplayedActions()}
+            onModalHide={() => goToNext()}
+            containerStyle={styles.modal}
+          >
+            {displayedActions === "lending" && account && (
+              <LendingBanners account={account} />
+            )}
+            {!!displayedActions &&
+              actions[displayedActions] &&
+              actions[displayedActions].map((a, i) =>
```

Comment:
could be as simple as `actions[displayedActions]?.map(...)` with optional chaining i think.